source/MaterialXRuntime/RtNameResolver.h

```diff
@@ -0,0 +1,126 @@
+/// replacements. This is for both to and from MaterialX resolution.
+///
+struct RtNameResolverInfo
+{
+    /// Element type enumeration
+    enum ElementType {
+        FILENAME_TYPE, /// Filename element type
+        GEOMNAME_TYPE  /// Geometry element type
+    };
+
+    RtNameResolverInfo() = default;
+    RtToken identifier {""}; /// Unique token identifier for this information
+    ElementType elementType; /// Type of element this resolver applies to
+    RtNameResolverFunction toFunction; /// Resolver function to resolve to MaterialX. Can be null.
+    RtNameResolverFunction fromFunction; /// Resolver function to resolve from MaterialX. Can be null.
+    RtTokenMap<RtToken> toSubstitutions; /// Custom token substitutions to MaterialX. May be empty.
+    RtTokenMap<RtToken> fromSubstitutions; /// Custom token substitutions from MaterialX. May be empty.
+};
+
+/// @class RtStringResolverPair
+///
+/// Class that keeps track of a pair of String Resolvers that are used to
+/// convert string identifiers to and from MaterialX.
+///
+class RtStringResolverPair
+{
+  public:
+    /// \brief Creates an RtStringResolverPair shared_ptr
+    /// \param type The type of element the resolver pair resolves
+    /// \param toMaterialXResolver Resolves string identifiers to MaterialX document format
+    /// \param fromMaterialXResolver Resolves string identifiers from MaterialX document format
+    /// \return Returns an RtStringResolverPair shared_ptr
+    static RtStringResolverPairPtr createNew(const RtNameResolverInfo::ElementType elementType,
+                                             MaterialX::StringResolverPtr toMaterialXResolver,
+                                             MaterialX::StringResolverPtr fromMaterialXResolver);
+
+    /// \brief Returns the type of element resolved by the pair of resolvers
+    /// \return String type of element resolved by the pair of resolvers
+    const RtNameResolverInfo::ElementType getType() const;
+
+    /// \brief Returns the resolver used to convert string identifiers to MaterialX document format
+    /// \return StringResolverPtr to the resolver used to convert string identifiers to MaterialX document format
+    MaterialX::StringResolverPtr getToMaterialXResolver();
+
+    /// \brief Returns the resolver used to convert string identifiers from MaterialX document format
+    /// \return StringResolverPtr to the resolver used to convert string identifiers from MaterialX document format
+    MaterialX::StringResolverPtr getFromMaterialXResolver();
+
+  private:
+    RtStringResolverPair(const RtNameResolverInfo::ElementType elementType,
+                         MaterialX::StringResolverPtr toMaterialXResolver,
+                         MaterialX::StringResolverPtr fromMaterialXResolver);
+
+    friend class PvtStringResolverPair;
+    std::unique_ptr<PvtStringResolverPair> _stringResolverPair;
+};
+
+/// @class RtNameResolverRegistrar
+class RtNameResolverRegistrar
```

Comment:
The registrar for name resolvers.

Reply:
Nitpick: For consistency we could use the word "registry" instead of "registrar", since that's the convention used everywhere else in the code base.